java/src/main/native/src/TableJni.cpp

```diff
@@ -2637,11 +2637,20 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_partition(JNIEnv *env, jc
     auto [partitioned_table, partition_offsets] =
         cudf::partition(*n_input_table, *n_part_column, number_of_partitions);
 
-    // for what ever reason partition returns the length of the result at then
-    // end and hash partition/round robin do not, so skip the last entry for
-    // consistency
     cudf::jni::native_jintArray n_output_offsets(env, output_offsets);
-    std::copy(partition_offsets.begin(), partition_offsets.end() - 1, n_output_offsets.begin());
+    // When row number is 0, the size of partition_offsets will be 0.
```

Comment:
This seems like a bug in libcudf that is better fixed there.  The documentation for cudf::partition states:
```
Returns a `vector<size_type>` of `num_partitions + 1` values
```

Reply:
I'll look into that.

Reply:
Maybe it is related to https://github.com/rapidsai/cudf/blob/branch-22.10/cpp/src/partitioning/partitioning.cu#L756